gcp_variant_transforms/libs/variant_sharding_test.py

```diff
@@ -0,0 +1,412 @@
+    self.assertEqual(sharder.get_shard('chr1', 1999999), 2)
+    # 'chr2' OR 'ch2' OR 'c2' OR '2'
+    self.assertEqual(sharder.get_shard('chr2', 0), 3)
+    self.assertEqual(sharder.get_shard('chr2', 999999999000), 3)
+    # '3:500,000-1,000,000'
+    self.assertEqual(sharder.get_shard('3', 500000), 4)
+    self.assertEqual(sharder.get_shard('3', 999999), 4)
+
+    # All the followings are assigned to residual shard.
+    self.assertEqual(sharder.get_shard('chr1', 2000000), 1)
+    self.assertEqual(sharder.get_shard('chr1', 999999999), 1)
+
+    self.assertEqual(sharder.get_shard('3', 0), 1)
+    self.assertEqual(sharder.get_shard('3', 499999), 1)
+    self.assertEqual(sharder.get_shard('3', 1000000), 1)
+
+    self.assertEqual(sharder.get_shard('ch2', 0), 1)
+    self.assertEqual(sharder.get_shard('c2', 0), 1)
+    self.assertEqual(sharder.get_shard('2', 0), 1)
+
+    self.assertEqual(sharder.get_shard('c4', 0), 1)
+    self.assertEqual(sharder.get_shard('cr5', 0), 1)
+    self.assertEqual(sharder.get_shard('chr6', 0), 1)
+
+  def test_config_residual_shard_absent(self):
+    sharder = variant_sharding.VariantSharding(
+        'gcp_variant_transforms/testing/data/sharding_configs/'
+        'residual_missing.yaml')
+    self.assertFalse(sharder.should_flatten())
+    self.assertEqual(sharder.get_num_shards(), 5)
+    # All shards excpet the last one (dummy residual) should be kept.
+    for i in range(sharder.get_num_shards() - 1):
+      self.assertTrue(sharder.should_keep_shard(i))
+    self.assertFalse(sharder.should_keep_shard(5 - 1))
+
+    # 'chr1:0-1,000,000'
+    self.assertEqual(sharder.get_shard('chr1', 0), 0)
+    self.assertEqual(sharder.get_shard('chr1', 999999), 0)
+    # 'chr1:1,000,000-2,000,000'
+    self.assertEqual(sharder.get_shard('chr1', 1000000), 1)
+    self.assertEqual(sharder.get_shard('chr1', 1999999), 1)
+    # 'chr2' OR 'ch2' OR 'c2' OR '2'
+    self.assertEqual(sharder.get_shard('chr2', 0), 2)
+    self.assertEqual(sharder.get_shard('chr2', 999999999000), 2)
+    # '3:500,000-1,000,000'
+    self.assertEqual(sharder.get_shard('3', 500000), 3)
+    self.assertEqual(sharder.get_shard('3', 999999), 3)
+
+    # All the followings are assigned to residual shard.
+    self.assertEqual(sharder.get_shard('chr1', 2000000), 4)
+    self.assertEqual(sharder.get_shard('chr1', 999999999), 4)
+
+    self.assertEqual(sharder.get_shard('3', 0), 4)
+    self.assertEqual(sharder.get_shard('3', 499999), 4)
+    self.assertEqual(sharder.get_shard('3', 1000000), 4)
+
+    self.assertEqual(sharder.get_shard('ch2', 0), 4)
+    self.assertEqual(sharder.get_shard('c2', 0), 4)
+    self.assertEqual(sharder.get_shard('2', 0), 4)
+
+    self.assertEqual(sharder.get_shard('c4', 0), 4)
+    self.assertEqual(sharder.get_shard('cr5', 0), 4)
+    self.assertEqual(sharder.get_shard('chr6', 0), 4)
+
+  def test_config_failed_missing_region(self):
+    tempdir = temp_dir.TempDir()
+    missing_region = [
+        '-  partition:',
```

Comment:
We still have partitions here, throughout the file.

Is it because we are not modifying the config file that gets generated?

Reply:
These are field names in the input YAML file:
https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/data/partition_configs/homo_sapiens_default.yaml

I decided not to modify those because we will replace that config file in the next PR.